addons/website/static/src/core/colibri.js

```diff
@@ -47,6 +47,31 @@ export class Colibri {
         if (!this.isStarted) {
             throw new Error("this.addListener can only be called after the interaction is started. Maybe move the call in the start method.");
         }
+        const re = /^(?<event>.*)\.(?<suffix>prevent|stop|capture|noupdate)$/;
+        let groups = re.exec(event)?.groups;
+        while (groups) {
+            fn = {
+                prevent: (f) => ((ev) => {
+                    ev.preventDefault();
+                    return f(ev);
+                }),
+                stop: (f) => ((ev) => {
+                    ev.stopPropagation();
+                    return f(ev);
+                }),
+                capture: (f) => {
+                    options ||= {};
+                    options.capture = true;
+                    return f;
+                },
+                noupdate: (f) => ((ev) => {
+                    f(ev);
+                    return SKIP_IMPLICIT_UPDATE;
+                }),
+            }[groups.suffix](fn);
+            event = groups.event;
```

Comment:
```js
while (groups) {
    switch (groups.suffix) {
        case "prevent":
            fn = (ev) => {
                ev.preventDefault();
                return fn(ev)
            };
            break;
        case "stop":
            fn = (ev) => {
                ev.stopPropagation();
                return fn(ev)
            };
            break;
        case "capture":
            options ||= {};
            options.capture = true;
            break;
        case "noupdate":
            fn = (ev) => {
                f(ev);
                return SKIP_IMPLICIT_UPDATE;
            };
            break;
        default:
            throw new Error("Invalid t-on- suffix (should be 'prevent', 'stop', 'capture', 'noupdate')");
        }
    }
    event = groups.event;
    groups = re.exec(event)?.groups;
}
```

Reply:
This then also requires to distinguish each depth function to avoir a recursive loop:
=>
```js
case ...: {
     let oldFn = fn;
     fn = (ev) => { .... oldFn(ev) ... }
     break;
}
```

TL;DR we keep the lookup object